aws/storagemanager/aws_test.go

```diff
@@ -0,0 +1,708 @@
+			//        Update from 2048 GiB to  4096 GiB by adding disks
+			request: &cloudops.StoragePoolUpdateRequest{
+				DesiredCapacity:     4096,
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				CurrentDriveSize:    1024,
+				CurrentDriveType:    "gp2",
+				CurrentDriveCount:   2,
+				TotalDrivesOnNode:   2,
+			},
+			response: &cloudops.StoragePoolUpdateResponse{
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				InstanceStorage: []*cloudops.StoragePoolSpec{
+					&cloudops.StoragePoolSpec{
+						DriveCapacityGiB: 1024,
+						DriveType:        "gp2",
+						DriveCount:       2,
+						IOPS:             3072,
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			// ***** TEST: 5
+			//		  Instances has 2 x 1024 GiB
+			//        Update from 2048 GiB to  3072 GiB by adding disks
+			request: &cloudops.StoragePoolUpdateRequest{
+				DesiredCapacity:     3072,
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				CurrentDriveSize:    1024,
+				CurrentDriveType:    "gp2",
+				CurrentDriveCount:   2,
+				TotalDrivesOnNode:   2,
+			},
+			response: &cloudops.StoragePoolUpdateResponse{
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				InstanceStorage: []*cloudops.StoragePoolSpec{
+					&cloudops.StoragePoolSpec{
+						DriveCapacityGiB: 1024,
+						DriveType:        "gp2",
+						DriveCount:       1,
+						IOPS:             3072,
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			// ***** TEST: 6
+			//		  Instances has 3 x 600 GiB
+			//        Update from 1800 GiB to 2000 GiB by adding disks
+			request: &cloudops.StoragePoolUpdateRequest{
+				DesiredCapacity:     2000,
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				CurrentDriveSize:    600,
+				CurrentDriveType:    "gp2",
+				CurrentDriveCount:   3,
+				TotalDrivesOnNode:   3,
+			},
+			response: &cloudops.StoragePoolUpdateResponse{
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				InstanceStorage: []*cloudops.StoragePoolSpec{
+					&cloudops.StoragePoolSpec{
+						DriveCapacityGiB: 600,
+						DriveType:        "gp2",
+						DriveCount:       1,
+						IOPS:             1800,
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			// ***** TEST: 7
+			//		  Instances has no existing drives
+			//        Update from 0 GiB to 700 GiB by adding disks
+			request: &cloudops.StoragePoolUpdateRequest{
+				DesiredCapacity:     700,
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				TotalDrivesOnNode:   0,
+			},
+			response: &cloudops.StoragePoolUpdateResponse{
+				ResizeOperationType: api.SdkStoragePool_RESIZE_TYPE_ADD_DISK,
+				InstanceStorage: []*cloudops.StoragePoolSpec{
+					&cloudops.StoragePoolSpec{
+						DriveCapacityGiB: 700,
+						DriveType:        "gp2",
+						DriveCount:       1,
+						IOPS:             2100,
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		/*{
```

Comment:
is this commented deliberately?

Reply:
this is the same test case which has been commented for all providers. The algorithm currently cannot handle adding disks to storage less nodes.